decompiler/structures/pseudo/expressions.py

```diff
@@ -250,6 +250,20 @@ def copy(self) -> ExternFunctionPointer:
         return ExternFunctionPointer(self.value, self._type.copy(), self.tags)
 
 
+class NotUseableConstant(Constant):
+    """Represents a non useable constant like 'inf' or 'NaN' as a string"""
+    def __init__(self, value: str, tags: Optional[Tuple[Tag, ...]] = None):
+        super().__init__(value, UnknownType(), tags=tags)
+
+    def __repr__(self):
+        """Return the non usable constant."""
+        return f"non useable constant {self.value}"
```

Comment:
Consistency with constant, perhaps something like the following is more natural:
```python
f"{self.value} type: not-usable-constant"
```

Reply:
Changed